pandas/io/parsers.py

```diff
@@ -384,6 +384,38 @@ def _validate_integer(name, val, min_val=0):
     return val
 
 
+def _check_dup_names(names):
+    """
+    Check if the `names` parameter contains duplicates.
+
+    Currently, this function issues a warning if that is the case. In the
+    future, we will raise an error.
+
+    Parameters
+    ----------
+    names : array-like or None
+        An array containing a list of the names used for the output DataFrame.
+    """
+
+    if names is None:
+        return
+
+    counts = {}
+    warn_dups = False
+
+    for name in names:
```

Comment:
just use set intersection here

Reply:
How so?  This is a fail-early method, which is why I chose it.

Reply:
simply check for ``len(names) !+ len(set(names)).``  much more idiomatic

Reply:
Fair enough.  Done.